Index: app/src/androidTest/java/de/monokel/frontend/LocalSaferTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/androidTest/java/de/monokel/frontend/LocalSaferTest.java	(date 1606736440854)
+++ app/src/androidTest/java/de/monokel/frontend/LocalSaferTest.java	(date 1606736440854)
@@ -0,0 +1,52 @@
+package de.monokel.frontend;
+
+import android.content.Context;
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.platform.app.InstrumentationRegistry;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import de.hhn.frontend.provider.LocalSafer;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test-class for the LocalSafer-Class
+ *
+ * @author Miftari
+ */
+@RunWith(AndroidJUnit4.class)
+public class LocalSaferTest {
+
+    Context appContext;
+
+    @Before
+    void setAppContext() {
+        appContext = InstrumentationRegistry.getInstrumentation().getTargetContext();
+    }
+
+    @Test
+    void settingsTest() {
+        LocalSafer.setIsAlarmRingLogged(false, appContext);
+        LocalSafer.setIsAlarmSetLogged(false, appContext);
+        LocalSafer.setIsKeySafeLogged(false, appContext);
+        LocalSafer.setIsKeyTransmitLogged(false, appContext);
+
+        assertFalse(LocalSafer.isAlarmRingLogged(appContext));
+        assertFalse(LocalSafer.isAlarmSetLogged(appContext));
+        assertFalse(LocalSafer.isKeySafeLogged(appContext));
+        assertFalse(LocalSafer.isKeyTransmitLogged(appContext));
+
+        LocalSafer.setIsAlarmRingLogged(true, appContext);
+        LocalSafer.setIsAlarmSetLogged(true, appContext);
+        LocalSafer.setIsKeySafeLogged(true, appContext);
+        LocalSafer.setIsKeyTransmitLogged(true, appContext);
+
+        assertTrue(LocalSafer.isAlarmRingLogged(appContext));
+        assertTrue(LocalSafer.isAlarmSetLogged(appContext));
+        assertTrue(LocalSafer.isKeySafeLogged(appContext));
+        assertTrue(LocalSafer.isKeyTransmitLogged(appContext));
+    }
+}
